### PR TITLE
Add support for https.

### PR DESCRIPTION
**Pull request by chiangf**
In order to use https as the protocol:
```
var mixpanel = Mixpanel.init('6fd9434dba686db2d1ab66b4462a3a67', { protocol: 'https' });
```

### CODE REVIEW

File: lib/mixpanel-node.js [hunk 1]
```diff
@@ -47,6 +49,8 @@ var create_client = function(token, config) {
             'verbose': metrics.config.verbose ? 1 : 0
         };
 
+        var protocol = metrics.config.protocol === 'http' ? http : https;
```
Comment (freeall) [inline lib/mixpanel-node.js]: I think it's better to throw if an unsupported protocol is given, instead of defaulting to anything.

Comment (chiangf) [inline lib/mixpanel-node.js]: Totally agree, changed in https://github.com/mixpanel/mixpanel-node/pull/96/commits/409c3e6c82f041ebf33dd2ad419c1914c20479f3

**Comment by chiangf** [comment]
Any thoughts on merging this PR in? I'd imagine that a lot of orgs require SSL

**Comment by tdumitrescu** [comment]
Well it is 2016 after all. Published in v0.5.0, thanks!

**Comment by builtbylane** [comment]
why isn't https the default?

**Comment by tdumitrescu** [comment]
No idea about the decisions that were made when this library was first started back in 2011. I do agree that at this point it should be the default, and could be convinced to make a v0.6.0 that switches it.

**Comment by freeall** [comment]
@tdumitrescu it should probably be a major version as I would think there's a possibility it may break in certain cases.

**Comment by chiangf** [comment]
Yup, the reason I made `http` as the default was so that the change didn't break for existing implementations.

**Comment by tdumitrescu** [comment]
That was my reasoning for wanting to defer to a v1 (see release notes for https://github.com/mixpanel/mixpanel-node/releases/tag/v0.5.0). At the same time though:
- semver permits breaking changes before 1.0.0
- a package.json spec with `^0.5.0` will not silently bump to 0.6.0: ["Allows changes that do not modify the left-most non-zero digit in the [major, minor, patch] tuple."](https://docs.npmjs.com/misc/semver#caret-ranges-123-025-004)

That said, either solution is fine with us.

**Pull request merged**